src/meta/src/rpc/service/hummock_service.rs

```diff
@@ -149,13 +151,20 @@ where
                 status: None,
             })),
             Some(mut compact_task) => {
-                self.hummock_manager
-                    .report_compact_task(
-                        req.context_id,
-                        &mut compact_task,
-                        Some(req.table_stats_change),
-                    )
-                    .await?;
+                let task_status = compact_task.task_status();
+                if CANCEL_STATUS_SET.contains(&task_status) {
+                    self.hummock_manager
+                        .cancel_compact_task(&mut compact_task, task_status)
+                        .await?;
```

Comment:
Can we ensure that all canceled tasks are not assigned? Because when a task is assigned, some states such as schedule policy and task heartbeat are changed, and these changes will be lost if `cancel_compact_task` is called on an assigned task.